### PR TITLE
Enable apache compression for static files

### PR DESCRIPTION
**Pull request by ncounter**
## What does this PR change?

Enable Apache compression for static files.
The pictures below (before/after) reports an example for just one file, but it is valid for all files with the following types:
- text/html
- text/plain
- text/xml
- text/css
- text/javascript
- application/javascript

Highlight on the `Transferred` value: the diff between the `size` value and the `transferred` one means how much network bandwidth we save downloading static files when they are compressed:
- `vendors.bundle.js` before: size = 834.43 KB , **transferred 835 KB**
- `vendors.bundle.js` after: size = 834.43 KB, **transferred 244.17 KB**

## GUI diff

Before:
not-compressed

After:
compressed

- [x] **DONE**

## Documentation
- No documentation needed: hidden technical detail

- [x] **DONE**

## Test coverage
- No tests: nothing to test

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/7887
Tracks # TODO

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"    
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle" (Test skipped, there are no changes to test)		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by srbarrios** [comment]
Hi Dario, I will try to figure out tomorrow how to unblock it, looks we need to somehow clean the check history. Thanks for your patience.

File: spacewalk/config/spacewalk-config.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Load modules to enable Apache static files compression
```
Comment (moio) [inline spacewalk/config/spacewalk-config.changes]:
```suggestion
- Load modules to enable Apache static file compression
```

Comment (ncounter) [reply spacewalk/config/spacewalk-config.changes]: Argh, typo fixed, thanks

**Comment by ncounter** [comment]
> General question: in the PR description you mention this only affect textual file types, but I see no mention of that in the proposed configs. Is that implicit, or are we compressing everything (including images, etc.)?

The [config code lines](https://github.com/uyuni-project/uyuni/pull/1608/files#diff-7caf6d814093cd7c87ff91863acfdaa6R176) to make it happen were already there, only modules were not enabled. About images, we are not using images in the product at all at the moment, but more important, it's not really recommended to compress images like `.jpg` or `.png` because they should already be and compressing them via the `deflate` module it just increase the server workload without producing benefits in the file size. Sometime it even happen that the file size increases.

Btw, I am updating this PR with two commits that I'll squash later if it gets approved, but I want you to take a look separately.
1. [first commit](https://github.com/uyuni-project/uyuni/pull/1608/commits/9efa309467302564ae177135152bccabf48e196b) I figured out that the `SetOutputFilter` was compressing everything by default and the `AddOutputFilterByType` was ignored. I removed that because for compressing a few file types only the proper rule to use is `AddOutputFilterByType`.
2. [second commit](https://github.com/uyuni-project/uyuni/pull/1608/commits/b1db160a58c932433deadfeeba5d4bab69abf604) I improved the `AddOutputFilterByType`: one important type was missing, the `.json` format. With that we can save **a lot** because many huge data are shaped and sent in a `.json` file at the moment. An example is the `/rhn/manager/admin/setup/products` Products page: the product data is a huge `.json` file of 1.25 MB, and the compression reduces it down to 74.30 KB only.

**Comment by ncounter** [comment]
@lneves12 :wink:

**Comment by moio** [comment]
OK, I dug up https://github.com/SUSE/spacewalk/pull/303

No idea why the modules were missing, apparently I even tested this.

Good to merge

**Pull request merged**